src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -242,18 +244,31 @@ protected void shadowUpdated(Map<String, Object> desired, Integer version) {
             return;
         }
         String configurationArn = configuration.getConfigurationArn();
+        if (configurationArn == null) {
+            logger.atError().log("Desired state has null configuration ARN. Ignoring shadow update");
+            return;
+        }
         boolean cancelDeployment = DESIRED_STATUS_CANCELED.equals(desired.get(DESIRED_STATUS_KEY));
 
         synchronized (ShadowDeploymentListener.class) {
+            if (lastConfigurationArn == null && reported != null && configurationArn
+                    .equals(reported.get(ARN_FOR_STATUS_KEY))) {
```

Comment:
Need clarification on a few more cases
1. If `cancelDeployment` and `lastConfigurationArn == null`, skip
1. If `!cancelDeployment` and `reported.get(STATUS_KEY) == in_progress`, should we still process the deployment, so the behavior is the same as jobs deployment? (If I understand correctly)
1. If `!cancelDeployment` and `reported.get(STATUS_KEY)` is a final state, skip

Reply:
Cleaned up the cases a bit. Case 2 I think we should still process it.